app/javascript/state/consent/index.js

```diff
@@ -1,103 +1,99 @@
+    case '@@chmp:consent:change_consent':
       return { ...state, consented: action.consented };
-    case '@@champaign:member:change_member_email':
-      return { ...state, email: action.email };
-    case '@@champaign:member:change_member_id':
-      return { ...state, memberId: action.memberId };
-    case '@@champaign:member:change_variant':
+    case '@@chmp:consent:change_variant':
       return { ...state, variant: action.variant };
-    case '@@champaign:member:reset_state':
+    case '@@chmp:consent:toggle_modal':
+      return { ...state, modalOpen: action.modalOpen };
+    case '@@chmp:consent:reset_state':
       return defaultState;
     default:
       return state;
   }
 }
 
-export function setPreviouslyConsented(previouslyConsented: boolean): Action {
-  return {
-    type: '@@champaign:member:change_previously_consented',
-    previouslyConsented,
-  };
-}
-
 export function changeConsent(consented: ?boolean = false): Action {
-  return { type: '@@champaign:member:change_consent', consented };
+  return { type: '@@chmp:consent:change_consent', consented };
 }
 
-export function changeCountry(countryCode: ?string = null): Action {
-  return { type: '@@champaign:member:change_country', countryCode };
-}
-
-export function changeMemberEmail(email: ?string = null): Action {
-  return { type: '@@champaign:member:change_member_email', email };
-}
-
-export function changeMemberId(memberId: ?string = null): Action {
-  return { type: '@@champaign:member:change_member_id', memberId };
+export function changeCountry(countryCode: string = ''): Action {
+  return { type: '@@chmp:consent:change_country', countryCode };
 }
 
 export function changeVariant(variant: string = 'simple'): Action {
-  return { type: '@@champaign:member:change_variant', variant };
+  return { type: '@@chmp:consent:change_variant', variant };
 }
 
 export function resetState(): Action {
-  return { type: '@@champaign:member:reset_state' };
+  return { type: '@@chmp:consent:reset_state' };
+}
+
+export function toggleModal(value: boolean): Action {
+  return { type: '@@chmp:consent:toggle_modal', modalOpen: value };
 }
 
-function isEU(countryCode: ?string, countries = window.champaign.countries) {
-  const country = countries.find(c => c.alpha2 === countryCode);
-  if (!country) return false;
-  return country.eea_member;
+// Conditions:
+// * selected country is in an affected country
+// * user has not previously given consent
+// * user has not selected to consent or not (`consented` is still null)
+function isRequired(state: ConsentState) {
+  const { countryCode, consented, previouslyConsented } = state;
+  // Affected countries: EEA members except Germany and Austria
+  const inAffectedCountry = includes(
+    window.champaign.countries
+      .filter(c => c.eea_member)
+      .filter(c => !includes(['DE', 'AT'], c.alpha2))
+      .map(c => c.alpha2),
+    countryCode
+  );
+  return inAffectedCountry && !previouslyConsented && consented === null;
```

Comment:
So this is not a user flow we present to existing members who _have_ previously consented but have then withdrawn their consent?

Reply:
🤔 I'm not sure actually. The way I understand the flow that was requested, we _always_ show this to existing members that have _not_ consented (by consented I mean said "Yes"). For instance, in the pop up, there's no option to say "No", but instead, a "Not right now" option which means they'll see the pop up the next time they log in.

If we make a slight change to `previouslyConsented === null`, it will only be shown to existing members who haven't made a choice yet.